src/char.rs

```diff
@@ -50,30 +54,124 @@ impl Char {
     }
 }
 
-impl From<Char> for i8 {
-    fn from(c: Char) -> i8 {
-        c.0
+impl From<Char> for char {
+    fn from(c: Char) -> char {
+        c.0 as u8 as char
+    }
+}
+
+#[doc(hidden)]
+impl FromGlib<c_char> for Char {
+    fn from_glib(value: c_char) -> Self {
+        Char(value)
+    }
+}
+
+#[doc(hidden)]
+impl ToGlib for Char {
+    type GlibType = c_char;
+
+    fn to_glib(&self) -> c_char {
+        self.0
+    }
+}
+
+/// Wrapper for values where C functions expect a plain C `unsigned char`
+///
+/// This `UChar` type is a wrapper over an `u8`, so that we can pass it to Glib or C functions.
+/// The check for whether a Rust `char` (a Unicode scalar value) actually fits in a `u8` is
+/// done in the `new` function; see its documentation for details.
+///
+/// The inner `u8` (which is equivalent to `libc::c_uchar` can be extracted with `.0`, or
+/// by calling `my_char.to_glib()`.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct UChar(pub u8);
+
+impl UChar {
+    /// Creates a `Some(UChar)` if the given `char` is representable as an `u8`
+    ///
+    /// # Example
+    /// ```ignore
+    /// extern "C" fn have_a_byte(b: libc::c_uchar);
+    ///
+    /// let a = Char::new('a').unwrap();
+    /// assert!(a.0 == 65);
+    /// have_a_byte(a.to_glib());
+    ///
+    /// let not_representable = Char::new('☔');
+    /// assert!(not_representable.is_none());
+    /// ```
+    pub fn new(c: char) -> Option<UChar> {
+        if c as u32 > 255 {
+            None
+        } else {
+            Some(UChar(c as u8))
+        }
+    }
+}
+
+impl From<UChar> for char {
+    fn from(c: UChar) -> char {
+        c.0 as char
+    }
+}
```

Comment:
Should there be `TryFrom` impls for the other direction? Or rather, with `TryFrom` still being unstable, the same function implemented in `impl Char` / `impl UChar`?

Reply:
IMHO `new` will be replaced with `TryFrom` when it stabilized.
Or you meant something other?

Reply:
Indeed, nevermind then :)